docs/DeveloperGuide.md

```diff
@@ -1,38 +1,591 @@
+* *membersFile*: A File object representing the `members.txt` file associated with the group.
+* *transactionsFile*: A File object representing the `transactions.txt` file associated with the group.
+* *members*: A MemberList object representing the list of Members in the group.
+* *transactions*: A TransactionList object representing the list of Transactions in the group.
+* *scanners*: A size 2 array of Scanners to be used for loading data from the data storage files. The first Scanner in the array is used for reading from `members.txt` while the second is used for reading from `transactions.txt`.
+
+<ins>Constructor</ins>
+
+The StorageHandler constructor creates the relevant data storage directories if they do not current exist while initializing the attributes of the object.
+
+Key arguments for the constructor are a `MemberList` object, a `TransactionList` object and a string `groupName`. The first two are used to represent the list of `Member` objects and the list of `Transaction` objects associated with the group for reference when loading or saving data. The last represents the directory to be written to to ensure that data across groups are kept discrete.
+
+<ins>Methods</ins>
+
+* *loadMembersData*: Reads data from `membersFile` and unpacks it before inserting `Member` objects into `MemberList`.
+* *loadTransactionsData*: Reads data from `transactionsFile` and unpacks it, checking if each member exists in `MemberList` before inserting `Transaction` objects into `TransactionList`.
+* *saveMembersData*: Writes packaged data from each `Member` and saves it as a record in `membersFile`.
+* *saveTransactionsData*: Writes packaged data from each `Transaction` and saves it as a record in `transactionsFile`.
+  
+Data loading methods are merged in the *loadAllData* method while data saving methods are merged in the *saveAllData* method.
+
+<ins>Usage Example</ins>
+
+![StorageHandler Sequence Diagram](diagrams/StorageHandlerSequenceDiagram.png)
+
+Given below is an example usage scenario and how StorageHandler behaves at each step:
+
+1. The user launches the application for the first time. Group is initialized, creating an instance of StorageHandler. StorageHandler creates relevant storage directories if they do not yet exist.
+2. StorageHandler reads data from the 2 data storage files and creates Member and Transaction objects in the associated utility list objects.
+3. When a command which would alter the data within MemberList or TransactionList is invoked, the method to save data to the storage files is called by Group. This updates the information within the storage files.
+
+<ins>Design Considerations</ins>
+
+* Update upon change, not upon exit - This allows for data to be saved even if the application exits ungracefully.
+* *checkTransactions* - Methods are provided to have a quick check to ensure that data from data storage is not corrupted.
+
+### Member and MemberList
```

Comment:
Good use of a sequence diagram, though consider adding a class diagram as well since the class functionality is rather complicated